lib/ohai/dsl/plugin.rb

```diff
@@ -183,6 +185,18 @@ def method_missing(name, *args)
 
       private
 
+      def safe_get_attribute(*keys)
+        keys.inject(@data) do |attrs, key|
+          unless attrs.nil? || attrs.is_a?(Hash)
```

Comment:
I don't think we want to allow Array access here. The idea is to prevent accessing values as subattribute keys. We expect the structure to be a Mash until we reach a value.